eth/beacon/state_machines/forks/serenity/validation.py

```diff
@@ -0,0 +1,251 @@
+from eth_typing import (
+    Hash32
+)
+from eth_utils import (
+    ValidationError,
+)
+import rlp
+from typing import Type
+
+from eth.constants import (
+    ZERO_HASH32,
+)
+
+
+from eth.beacon.enums import (
+    SignatureDomain,
+)
+from eth.beacon.helpers import (
+    get_attestation_participants,
+    get_block_root,
+    get_domain,
+)
+from eth._utils import (
+    bls
+)
+from eth.beacon._utils.hash import (
+    hash_eth2,
+)
+from eth.beacon.types.states import BeaconState  # noqa: F401
+from eth.beacon.types.attestations import Attestation  # noqa: F401
+from eth.beacon.types.attestation_data import AttestationData  # noqa: F401
+
+
+#
+# Attestation validation
+#
+def validate_serenity_attestation(state: Type[BeaconState],
+                                  attestation: Attestation,
+                                  epoch_length: int,
+                                  min_attestation_inclusion_delay: int,
+                                  is_validating_signatures: bool=True) -> None:
```

Comment:
Is `is_validating_signatures` for the convenience of testing?

Reply:
It was in the deprecated version of this validation. I think there might be some cases that this is useful in which you might be processing a block that you created with attestations you already checked the signatures of individually and aggregated.

That said, we should only add this in when we actually run into a concrete usecase.

As for speeding up tests, we should do that in a more systematic monkeypatch if we have a suite of tests that we want to speed up without validation.